pkg/visitors/nvramcompact.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 the LinuxBoot Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package visitors
+
+import (
+	"github.com/linuxboot/fiano/pkg/guid"
+	"github.com/linuxboot/fiano/pkg/uefi"
+)
+
+func compactNVarStore(s *uefi.NVarStore) error {
+	var keepEntries []*uefi.NVar
+	linkedNVar := make(map[uint64]*uefi.NVar)
+	// Find Data entries and associated metadata entries
+	for _, v := range s.Entries {
```

Comment:
https://github.com/linuxboot/linuxboot/blob/master/bin/nvram-compact seems to imply that the name is stored on the first of the links to the nvar variable, while the last contains the actual value. Would we lose the name of the nvar in this case?

Reply:
Yes the first has the name, GUID (plain or offset in the guidstore) and possible content.
The next links are updates to the value (like a journal).
During the parsing the name and GUID are copied in the NVar go struct (see: https://github.com/linuxboot/fiano/blob/master/pkg/uefi/nvram.go#L358 ) so we don't need to copy them in compact, only just select the last one.

Then we rebuild a `FullNVarEntry` that contains name, GUID and content.

Reply:
I see! ok